lib/v2.recommended.yaml

```diff
@@ -0,0 +1,231 @@
+    prefer_final_locals: warning
+    prefer_function_declarations_over_variables: warning
+    prefer_if_elements_to_conditional_expressions: warning
+    prefer_initializing_formals: warning
+    prefer_inlined_adds: warning
+    prefer_int_literals: warning
+    prefer_interpolation_to_compose_strings: warning
+    prefer_iterable_whereType: warning
+    prefer_mixin: warning
+    prefer_null_aware_operators: warning
+    prefer_single_quotes: warning
+    prefer_spread_collections: warning
+    prefer_void_to_null: warning
+    provide_deprecation_message: warning
+    sort_pub_dependencies: warning
+    sort_unnamed_constructors_first: warning
+    test_types_in_equals: warning
+    throw_in_finally: warning
+    type_annotate_public_apis: warning
+    unawaited_futures: warning
+    unnecessary_await_in_return: warning
+    unnecessary_brace_in_string_interps: warning
+    unnecessary_cast: warning
+    unnecessary_getters_setters: warning
+    unnecessary_lambdas: warning
+    unnecessary_parenthesis: warning
+    unnecessary_overrides: warning
+    unnecessary_this: warning
+    use_function_type_syntax_for_parameters: warning
+    use_rethrow_when_possible: warning
+    use_setters_to_change_properties: warning
+    use_string_buffers: warning
+    use_to_and_as_if_applicable: warning
+
+linter:
+  rules:
+    - always_declare_return_types
+    - always_put_required_named_parameters_first
+    - always_require_non_null_named_parameters
+    - avoid_bool_literals_in_conditional_expressions
+    - avoid_field_initializers_in_const_classes
+    - avoid_function_literals_in_foreach_calls
+    - avoid_implementing_value_types
+    - avoid_js_rounded_ints
+    - avoid_null_checks_in_equality_operators
+    - avoid_positional_boolean_parameters
+    - avoid_relative_lib_imports
+    - avoid_renaming_method_parameters
+    - avoid_returning_this
+    - avoid_returning_null
+    - avoid_returning_null_for_future
+    - avoid_returning_null_for_void
+    - avoid_setters_without_getters
+    - avoid_slow_async_io
+    - avoid_types_on_closure_parameters
+    - avoid_unused_constructor_parameters
+    - cascade_invocations
+    - comment_references
+    - control_flow_in_finally
+    - curly_braces_in_flow_control_structures
+    - directives_ordering
+    - duplicate_ignore
+    - invalid_required_named_param
+    - invalid_required_positional_param
+    - invariant_booleans
+    - join_return_with_assignment
+    - literal_only_boolean_expressions
+    - omit_local_variable_types
+    - overridden_fields
+    - package_api_docs
+    - package_prefixed_library_names
+    - package_names
+    - prefer_adjacent_string_concatenation
+    - prefer_asserts_in_initializer_lists
+    - prefer_collection_literals
+    - prefer_conditional_assignment
+    - prefer_const_constructors_in_immutables
+    - prefer_const_declarations
+    - prefer_const_literals_to_create_immutables
+    - prefer_constructors_over_static_methods
```

Comment:
When we looked at this previously, we decided it was too strict and that in some cases, static methods were better. I unfortunately don't have any more info than that. Do you happen to know if this lints async static methods?

Reply:
Quick test shows that it does not
```
class TestThing {
  final bool _testBool;

  TestThing(this._testBool);

  // triggers prefer constructor lint
  static TestThing getter(bool value) => TestThing(value);

  // no lint
  static Future<TestThing> getterAsync(bool value) async => TestThing(value);
}
```

Reply:
Yup I think so, thanks!